src/core/config/sources/query.js

```diff
@@ -9,7 +10,16 @@ import { parseSearch } from "core/utils"
  */
 
 const optionsFromQuery = () => (options) => {
```

Comment:
How about the following?

Pros:

- the exception of `urls.primaryName` is handled explicitly
- for any other case `set` is used
- we're using reduction on empty `{}` instead of mutating and deleting from `queryOptions`

```js
const optionsFromQuery = () => (options) => {
  const urlSearchParams = options.queryConfigEnabled ? parseSearch() : {}

  return Object.entries(urlSearchParams).reduce((acc, [key, value]) => {
    if (key === 'urls.primaryName') {
       acc[key] = value;
    } else {
       acc = set(queryOptions, key, value)
    }
    return acc;
  }, {});
}
```